wallet-api/src/Wallet/API.hs

```diff
@@ -269,6 +283,34 @@ class WalletAPI m where
     -}
     slot :: m Slot
 
+-- | Change the 'validationInterval' of a 'WalletAPI' action.
+newtype WithValidationInterval m a = WithValidationInterval { runWithValidationInterval :: m (Slot, Slot) -> m a }
```

Comment:
@michaelpj I'm thinking about the following change to `WalletAPI`:

* Add a type `WalletAPIConfig { validationInterval :: Slot -> (Slot, Slot), coinSelection :: CoinSelection, feePolicy :: FeePolicy, ownKeyPair :: KeyPair }` with everything that the user might want to change for specific transactions
* Remove those fields from `WalletAPI`
* Change wallet actions to `(MonadReader WalletAPIConfig m, WalletAPI m)`

Then we do things like `withValidationInterval` below in a more principled way. What do you think?